vendor/github.com/coreos/go-oidc/v3/oidc/jwks.go

```diff
@@ -120,11 +111,10 @@ func (r *remoteKeySet) verify(ctx context.Context, jws *jose.JSONWebSignature) (
 		}
 	}
 
-	if !r.now().Add(keysExpiryDelta).After(expiry) {
-		// Keys haven't expired, don't refresh.
-		return nil, errors.New("failed to verify id token signature")
-	}
-
+	// If the kid doesn't match, check for new keys from the remote. This is the
+	// strategy recommended by the spec.
+	//
+	// https://openid.net/specs/openid-connect-core-1_0.html#RotateSigKeys
 	keys, err := r.keysFromRemote(ctx)
```

Comment:
does this open us up to DOS via random key id in presented tokens?

Reply:
previously there was a 30 second [delta](https://github.com/kubernetes/kubernetes/pull/123402/files#diff-21bdef324de021527b14e3f61ed5085eecd38352385b14a0ab8e8391b1fb1c29L24) but with this change, yes every random kid would result in a call to the jwks endpoint.

Reply:
Chatted with @enj offline and he suggested avoiding the dep bump and instead setting `SkipIssuerCheck=true`, performing the issuer validation in authenticator (similar to audience check) when discovery URL is set. I'll make that change in a separate PR and close this after.

Reply:
opened https://github.com/kubernetes/kubernetes/pull/123527